server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaTokenStore.java

```diff
@@ -298,10 +303,10 @@ public String getUserId() {
         }
 
         public boolean isExpired() {
-            if (getExpiresAt()==0) {
-                return new Timestamp(System.currentTimeMillis()-getExpirationTime()).after(getCreated());
+            if (getExpiresAt().isEmpty()) {
+                return Instant.now().minus(getExpirationTime()).isAfter(getCreated());
             } else {
-                return getExpiresAt() < System.currentTimeMillis();
+                return getExpiresAt().get().isBefore(Instant.now());
```

Comment:
Sonar is happy if you do

        public boolean isExpired() {
            Optional<Instant> optionalExpiresAt = getExpiresAt();
            if (optionalExpiresAt.isEmpty()) {
                return Instant.now().minus(getExpirationTime()).isAfter(getCreated());
            } else {
                return optionalExpiresAt.get().isBefore(Instant.now());
            }
        }